### PR TITLE
Updates for TM 7.7.1 + big refactoring

### PR DESCRIPTION
**Pull request by davidwaroquiers**
Switch to src layout.
Moved tests outside of src code directory.
Switch from poetry to setuptools.

### CODE REVIEW

**Comment by davidwaroquiers** [comment]
@gpetretto This should be ready for review. I spent some time in the refactoring of the tests, as well as adding unit tests so that coverage is 100% (of course with some no cover/no branch here and there, but each time with a - usually provided - good reason).

File: pyproject.toml [hunk 1]
```diff
@@ -1,44 +1,63 @@
...
+    "Operating System :: OS Independent",
+    "Topic :: Other/Nonlisted Topic",
+    "Topic :: Scientific/Engineering",
+]
+requires-python = ">=3.9"
+dependencies =[
+    "numpy >= 1.16.5",
+    "matplotlib >= 3.5.1",
+    "pandas >= 1.4.1",
+    "pymatgen >= 2022.2.10",
```
Comment (gpetretto) [inline pyproject.toml:37]: Does everything work with the old 2022.2.10 version? The tests here run only with the latest. I suppose we should keep this for compatibility with old workflows

**Review by gpetretto — APPROVED** [review]
Thanks for the huge work. I just added a few comments, but I don't think there is any blocking issue.

File: src/turbomoleio/testing/__init__.py [hunk 2]
```diff
@@ -92,6 +88,35 @@
...
+    if len(found_paths) > 1:
+        raise RuntimeError("Found multiple turbomoleio directories.")
+    return found_paths[0]
+
+
+def get_tests_configs_tm_versions(test_data_dir):
+    """Get the test configs for all supported tm versions."""
+    from monty.serialization import loadfn
+
+    from turbomoleio.testing import TM_VERSIONS
```
[the remark this answers is not in the document]
Comment (davidwaroquiers) [reply src/turbomoleio/testing/__init__.py:112]: Correct. I removed

File: src/turbomoleio/testing/__init__.py [hunk 3]
```diff
@@ -92,6 +88,35 @@
 ]
 
 
+def get_test_data_dir(module=None):
+    """Dynamically and recursively look for the test_data directory."""
+    module = module or __file__
+    p = Path(module)
+    parts = p.parts
+    found_paths = []
+    for ii in range(len(parts), 0, -1):
```
[the remark this answers is not in the document]
Comment (davidwaroquiers) [reply src/turbomoleio/testing/__init__.py:97]: I explained what's done. Also added an additional unit test for it.

File: tests/core/test_base.py [hunk 4]
```diff
@@ -0,0 +1,151 @@
+# -*- coding: utf-8 -*-
+# The turbomoleio package, a python interface to Turbomole
+# for preparing inputs, parsing outputs and other related tools.
+#
+# Copyright (C) 2018-2022 BASF SE, Matgenix SRL.
```
[the remark this answers is not in the document]
Comment (davidwaroquiers) [reply tests/core/test_base.py:5]: I'm not sure it is actually needed but I'll check with our lawyer. In any case for open source it's not super important anyway. For now I'll keep it as such.

File: pyproject.toml (hunk 1 above)
Comment (davidwaroquiers) [reply pyproject.toml:37]: no you're right and it probably does not work with that older version... I'll check and update to a more recent that works.

Comment (davidwaroquiers) [reply pyproject.toml:37]: Well it seems it's working. So I'll keep it with 2022.

**Pull request merged**